tqecjs/src/workspace/plaquettes/PlaquetteClass.js

```diff
@@ -0,0 +1,370 @@
+	};
+
+	_ccw = (p, q, r) => {
+		// Check if the points are counterclockwise
+		const v1 = { x: q.globalX - p.globalX, y: q.globalY - p.globalY };
+		const v2 = { x: r.globalX - q.globalX, y: r.globalY - q.globalY };
+		// Take the cross product
+		const val = -(v1.x * v2.y - v1.y * v2.x); // If val > 0, then counterclockwise, else clockwise
+		if (val === 0) return 0; // Collinear
+		return val; // Clockwise  < 0 or counterclockwise >0
+	};
+
+	createPlaquette = () => {
+		// The graphic should connect the points from the qubits
+		const nQubits = this.qubits.length;
+		if (nQubits < 3) {
+			console.log('Plaquette must have at least 3 qubits');
+			this.plaquetteMade = false;
+			return null;
+		}
+
+		// Create a convex hull
+		this._createConvexHull();
+		const { x, y } = this.calculatePlaquetteCenter();
+		// Set the position and pivot of the plaquette
+		this.pivot.set(x, y);
+		this.position.set(x, y);
+		// Allow for the object to be interactive
+		this.plaquetteMade = true;
+		this.interactive = true;
+		this.cursor = 'pointer';
+		this.makeExtensible();
+		this.toggleCtrlButtons();
+	};
+
+	makeExtensible = () => {
+		// Make the plaquette extensible
+		this.interactive = true;
+		this.buttonMode = true;
+		this.on('pointerdown', this.onDragStart);
+		this.on('pointermove', this.onDragMove);
+		this.on('pointerup', this.onDragEnd);
+	};
+	onDragStart(event) {
+		this.isDragging = true;
+		this.initialPosition = event.data.getLocalPosition(this.parent);
+		console.log(this.initialPosition);
+	}
+
+	onDragMove = (event) => {
+		let newQubits = [];
+		let diff = 0;
+		// Get current position
+		const newPosition = event.data.getLocalPosition(this.parent);
+		// Calculate the distance moved
+		const deltaX = newPosition.x - this.initialPosition?.x;
+		const deltaY = newPosition.y - this.initialPosition?.y;
+		console.log(deltaY, deltaY);
+		let shiftQ = null;
+		if (
+			this.isDragging &&
+			(deltaX >= this.gridSize * 0.4 || deltaY >= this.gridSize * 0.4)
+		) {
+			// Check which direction the plaquette is being dragged
+			if (Math.abs(deltaX) > Math.abs(deltaY)) {
+				// Dragging horizontally
+				if (deltaX > 0) {
+					// Moving to the right
+					shiftQ = this.mostRightQubit();
+					// Find the neighboring qubit that is closest to the right
+					const newrq = shiftQ.neighbors.find(
+						(qubit) =>
+							qubit.globalX > shiftQ.globalX && qubit.globalY === shiftQ.globalY
+					);
+
+					diff = newrq.globalX - shiftQ.globalX;
+				} else {
+					// Generate the qubits that are closest to the left
+					shiftQ = this.mostLeftQubit();
+					// Find the neighboring qubit that is closest to the left
+					const newlq = shiftQ.neighbors.find(
+						(qubit) =>
+							qubit.globalX < shiftQ.globalX && qubit.globalY === shiftQ.globalY
+					);
+					diff = newlq.globalX - shiftQ.globalX;
+				}
+				// Shift the qubits by the difference
+				for (const qubit of this.qubits) {
+					const q = qubit.neighbors.find(
+						(q) =>
```

Comment:
You can add `// eslint-disable-next-line` here to suppress a warning about `diff`

Apparently ESlint dislikes this pattern -- https://stackoverflow.com/questions/58816244/debugging-eslint-warning-function-declared-in-a-loop-contains-unsafe-reference

Reply:
I believe that specific annotation needs to be on the line with `(q) =>`